include/onnx-mlir/Runtime/OMSignature.h

```diff
@@ -19,18 +19,20 @@ extern "C" {
 #endif
 
 /**
- * \brief Return the model's input signature as a JSON string.
+ * \brief Return the input signature of the given entry point as a JSON string.
  *
+ * @param entry point name
  * @return pointer to input signature JSON string
  */
-const char *omInputSignature();
+const char *omInputSignature(const char *entryPointName);
```

Comment:
Can you briefly explain how the entryPointName is used to locate the right signature? I tried to read the LLVM changes but I could not really get the big picture.

Reply:
LLVM code is equivalent to the following pseudo code:
```C
entry_point_0 = "run_add";
entry_point_0_in_sig = "[....]"; // JSON string.
entry_point_1 = "run_sub";
entry_point_1_in_sig = "[....]"; // JSON string.
const char* omInputSignature(entry_point) {
    if (entry_point == entry_point_1)
      return entry_point_0_in_sig;
    if (entry_point == entry_point_1)
      return entry_point_1_in_sig;
    return undefined;
}
```

We do not see if-then-else but branches at LLVM level.

Reply:
Cool, and the compare `entry_point == entry_point_0` simply compare the addresses of the strings?

Reply:
Yes, the address of the first char.

Reply:
This seems at odd with your comment that the comparison is done on the string content. Can you run a test to make sure that works? Thanks